src/sb-util.ts

```diff
@@ -1,3 +1,49 @@
-export function sb() {
-  console.log('sb-util')
-}
\ No newline at end of file
+import { Queryable, ScratchProjectOptions } from './abstracts';
+import { SpOptionsEmptyError, SpMultipleSourceOptionsError } from './errors';
+import { ProjectSource, AssetFetcher } from '../common';
+
+export class ScratchProject implements Queryable {
+	constructor(projectJSON: Object, assetFetcher?: any) {
+
+	}
+
+	query(selector?: String) {
+		return [];
+	}
+}
+
+/*
+	initialize() is a factory method that returns a
+	Promise to a ScratchProject while it does the
+	necessary I/O or HTTP requests needed to populate
+	the ScratchProject
+*/
+export const initialize = function(options: ScratchProjectOptions): Promise<ScratchProject> {
+	// Check project options before initializing a ScratchProject
+	const numProjectOptions = Object.keys(options).length;
+	if (!options ||  numProjectOptions === 0) { 
```

Comment:
Long term thing, validate the options in their own function. That would be easier to test (and faster).